content/blog/parsnip-checking/index.Rmd

```diff
@@ -0,0 +1,361 @@
+```
+
+Note the ending of the message: "...using the `rpart` engine." We didn't specify that we wanted to use `rpart` as an engine, yet that seems to be what went wrong!
+
+Readers who have fitted bagged decision tree models with parsnip before may realize that `rpart` is the default engine for these models. This shouldn't be requisite knowledge to interpret this message, though, and is not helpful in addressing the issue. With v1.0.2, we only mention the information that users give to us when constructing that message, and tell them exactly which packages they might need to install/load:
+
+``` r
+library(parsnip)
+
+bag_tree()
+#> ! parsnip could not locate an implementation for `bag_tree` model
+#>   specifications.
+#> ℹ The parsnip extension packages censored and baguette implement support for
+#>   this specification.
+#> ℹ Please install (if needed) and load to continue.
+#> 
+#> Bagged Decision Tree Model Specification (unknown mode)
+#> 
+#> Main Arguments:
+#>   cost_complexity = 0
+#>   min_n = 2
+#> 
+#> Computational engine: rpart
+```
+
+We hinted at another example of this guideline in the previous section; parsnip shouldn't refer to internal functions when it raises error messages. Above, with parsnip v1.0.1, we saw:
+
+``` r
+library(censored)
+#> Loading required package: survival
+
+bag_tree() %>%
+  set_engine("rpart") %>%
+  set_mode("regression")
+#> Error in `stop_incompatible_mode()`:
+#> ! Available modes for engine rpart are: 'unknown', 'censored regression'
+```
+
+The error points out a function called `stop_incompatible_mode()`, which is a function used internally by parsnip to check modes. There's a different function, `check_spec_mode_engine_val()`, that will flag super silly modes:
+
+``` r
+library(parsnip)
+
+bag_tree() %>%
+  set_engine("rpart") %>%
+  set_mode("beep bop boop")
+#> Error in `check_spec_mode_engine_val()`:
+#> ! 'beep bop boop' is not a known mode for model `bag_tree()`.
+```
+
+The important part, though, is that _the technical reasons don't matter_. Users don't know---and don't need to know---what `stop_incompatible_mode()` or `check_spec_mode_engine_val()` do.
+
+In parsnip v1.0.2, we now point users to the function they actually called that eventually gave rise to the error:
+
+``` r
+bag_tree() %>%
+  set_engine("rpart") %>%
+  set_mode("beep bop boop")
+#> Error in `set_mode()`:
+#> ! 'beep bop boop' is not a known mode for model `bag_tree()`.
+```
+
+We hope these changes improve folks' experience when modeling with parsnip in the future!
+
+## Other bits and bobs
+
+<!-- This post has highlighted upcoming improvements to model specification checking in parsnip. For those who'd like to learn more, I've written a [companion article](https://simonpcouch.com/blog) on my blog that delves further into the tooling we use to check model specifications. -->
+
+parsnip v1.0.2 includes a number of other changes that you can read about [here](https://github.com/tidymodels/parsnip/blob/main/NEWS.md).
```

Comment:
I think this is getting lost a bit here. I would keep this as the usual thank you section and move that link up to the first section. Something like "this post focuses on xyz but you can see the full [release notes](link to the release notes of specifically this version, once the pkgdown site has been updated) for all other changes".

Reply:
Addressed in [e5f39c4](https://github.com/tidyverse/tidyverse.org/pull/597/commits/e5f39c492bc4d37322e94efeaef522b4f34f2c3e). :)